utils/release/git-lock.mjs

```diff
@@ -29,6 +38,18 @@ const ensureUpToDateBranch = async () => {
   }
 };
 
+/**
+ * This will make .github\workflows\git-lock-fail.yml run and thus fail the associated check.
+ */
+const lockBranch = async () => {
```

Comment:
So the purpose of this is to create a failed check that will prevent merging ?

Doesn't github have a way to setting that controls admins from bypassing checks?

Guessing this is not related a concurrency issue that could be solved using [Concurrency Workflow syntax for GitHub Actions - GitHub Docs](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#concurrency) ?

Reply:
> So the purpose of this is to create a failed check that will prevent merging?

Exactly, because as long as we have to write access to the repo, if we pass the check we will be able to merge 'green' PR.
So I need to paint it red, because if I remove our write access... Well I have no means to reestablish them easily in case something goes wrong.

> Doesn't github have a way to setting that controls admins from bypassing checks?

Yep, and we enable this setting here:
https://github.com/coveo/cli/blob/c0b6610e156efed43cdb80d858333047e8f1b434/utils/release/lock-master.mjs#L38-L39

> Guessing this is not related a concurrency issue that could be solved using [Concurrency Workflow syntax for GitHub Actions - GitHub Docs](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#concurrency) ?

No indeed, this cannot be resolved by concurrency for several reasons:
 1. If a PR has already passed the checks, it will be able to go through, concurrency wouldn't help prevent that.
 2. If we setup a concurrency constraint on the CI, that means that only one PR can run its checks at a time, this would slow us down